dipy/tracking/localtrack.pyx

```diff
@@ -56,14 +56,21 @@ def local_tracker(
     cdef:
         cnp.npy_intp i
         StreamlineStatus stream_status
+        double dir[3]
+        double vs[3]
+        double seed[3]
 
     if (seed_pos.shape[0] != 3 or first_step.shape[0] != 3 or
             voxel_size.shape[0] != 3 or streamline.shape[1] != 3):
         raise ValueError('Invalid input parameter dimensions.')
 
+    for i in range(3):
+        dir[i] = first_step[i]
+        vs[i] = voxel_size[i]
+        seed[i] = seed_pos[i]
```

Comment:
Why this copy is needed? I do not get it .

We might need to rename `dir` since this is a python keyword

Reply:
if copy is needed, I will use some like:

```python
# Memory copy 
memcpy(dir, &first_step[0], sizeof(double) * 3)
```

you also have the `copy_point` function in `dipy.utils.fast_numpy`

`copy_point(&first_step[0], dir)`

Reply:
Yes, thanks, I will update the code. 

We need copies because `generate_streamline(.)` can modify them and the input values are further needed in `_generate_tractogram(.)`.  The bug here was that `dir` was modified with the current tracking orientation in `generate_streamline(.)`, and later used as input for the computation of the backward streamline segment in `generate_tractogram(.)`. It was supposed to be the initial orientation, but instead was the last tracking orientation. Making copies solves the issue.